components/nav.js

```diff
@@ -66,17 +142,125 @@ export const NavButton = ({ color = 'red', sx, ...props }) => (
         boxShadow: '0 0 0 2px',
         outline: 'none'
       },
-      ...sx
+      ...csx
     }}
   />
 )
 
+const IconWrapper = ({ csx, children, ...props }) => {
+  return (
+    <motion.div
+      {...props}
+      sx={{
+        display: 'inline-flex',
+        padding: '4px',
+        color: 'red',
+        borderRadius: 'circle',
+        textDecoration: 'none',
+        mr: [3, 4],
+        cursor: 'pointer',
+        transition: 'box-shadow .125s ease-in-out',
+        ':hover,:focus': {
+          boxShadow: '0 0 0 2px',
+          outline: 'none'
+        },
+        ...csx
+      }}
+    >
+      {children}
+    </motion.div>
+  )
+}
+
+const SearchBar = ({ setVisible, visible, search, ...props }) => {
+  const inp = useRef(null)
+
+  const handle_focus = () => {
+    const display_val = inp.current.style.display
+
+    if (display_val == 'none') inp.current.blur()
+    else inp.current.focus()
+  }
+
+  const onClick = () => {
+    setVisible(!visible)
+  }
+
+  const retAnim = visible => {
+    if (visible) {
+      console.log(window.innerWidth <= 512 ? 'mobVisible' : 'visible')
+      return window.innerWidth <= 512 ? 'mobVisible' : 'visible'
+    }
+    return 'hidden'
+  }
+
+  return (
+    <motion.div
+      variants={parentVariant}
+      animate={retAnim(visible)}
+      sx={{
+        display: 'flex',
+        borderRadius: 'circle',
+        py: [1]
+      }}
+      layout
+      onAnimationComplete={handle_focus}
+    >
+      <IconWrapper
+        variants={childVariant}
+        animate={visible ? 'hidden' : 'visible'}
+        onClick={onClick}
+      >
+        <Search size={24} />
+      </IconWrapper>
+
+      <motion.input
```

Comment:
Could this input be more inline with the Hack Club theme?

Specifically having less border radius and changing based on light/dark mode. Happy to do this if needed.

Reply:
Hey Sam 👋  

So, I would highly encourage you in helping with the dark theme colors for the input tag, Do you have any suggestion on what color it can be?

For the border-radius:

so if we reduce the border radius then this is what it will look like:

![image](https://user-images.githubusercontent.com/41180869/118927793-f053fd80-b95f-11eb-921e-d22a2d99c55c.png)

This is why I preferred it bo more rounded

But if you think that having it less rounded makes it look better then do lemme know :)

Reply:
If you're using CSS I think `var(--theme-ui-colors-background)` would work best

I personally prefer the less rounded variant though I acknowledge these things are highly subjective, @maxwofford @MatthewStanciu thoughts?